opts/opts.go

```diff
@@ -54,6 +58,31 @@ type ListOpts struct {
 	validator ValidatorFctType
 }
 
+//MapOpts type
+type MapOpts struct {
+	values *map[string]string
+}
+
+func (opts *MapOpts) Set(value string) error {
+	vals := strings.Split(value, "=")
```

Comment:
Just checked runconfig/parse.go, and it does splitN with n=2. For consistency between `docker run` and `docker -d`, i'll make it splitN